tests/datasets/test_utils.py

```diff
@@ -537,3 +542,101 @@ def test_azcopy(tmp_path: Path, azcopy: Executable) -> None:
 def test_which() -> None:
     with pytest.raises(DependencyNotFoundError, match='foo is not installed'):
         which('foo')
+
+
+def create_test_raster_file(
```

Comment:
Code used during testing usually belongs as a fixture, either in here or in `conftests.py`